ui/src/earn/providers/base.ts

```diff
@@ -0,0 +1,74 @@
+import type { ChainConfig } from '../../types/chainConfig';
+import type { WalletClient } from 'viem';
+import type { Avalanche } from '@avalanche-sdk/chainkit';
+import type { EarnPool } from '../types';
+
+/**
+ * Abstract base interface for earn providers
+ * All earn providers (AAVE, Benqi, etc.) must implement this interface
+ */
+export interface EarnProviderBase {
```

Comment:
I think we can define the type for the params and returned values for most of the methods in a separate types file And use those here. Also we expose a Address type from client (reexport of viem), which can be used instead of `0x${string}`

Reply:
I think you already have it, just need to move the type definitions here `ui/src/earn/types.ts`